### PR TITLE
Feature/sole parent support

### PR DESCRIPTION
**Pull request by verbman**
Thanks for contributing to OpenFisca !

* Tax and benefit system evolution.
* Impacted periods: Social Security
* Impacted areas: `openfisca-aotearoa/variables/social_security`
* Details:
  - Sole Parent Support benefit

- - - -

These changes _(remove lines which are not relevant to your contribution)_:

- Impact the OpenFisca-Aotearoa public API
- Add a new feature
- Fix or improve an already existing calculation.
- Change non-functional parts of this repository (for instance editing the README)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,58 @@
 # Changelog
 
+# 17.2.0 - [42](https://github.com/govzeroaotearoa/openfisca-aotearoa/pull/42)
```
Comment (bonjourmauko) [inline CHANGELOG.md]: Do not forget to bump also the `setup.py` file.

File: openfisca_aotearoa/entities.py [hunk 2]
```diff
@@ -65,7 +65,17 @@
             The one or more persons who are partners of the principal.
             Utilise variables to further define if a person fits the legal definition of a partner as it differs between legal instruments.
             In the case of the Social Security Act this can be considered an alternative to "social_security__in_a_relationship"
-            """
+            """,
+            },
+        {
```
Comment (bonjourmauko) [inline openfisca_aotearoa/entities.py:70]: This is exclusive with principal, right?

File: openfisca_aotearoa/tests/social_security/sole_parent_support/sole_parent_support__split_care.yaml [hunk 3]
```diff
@@ -0,0 +1,303 @@
+- name: Sole Parent Split Care, no one eligible - why is this false and not incalculatable?
```
Comment (bonjourmauko) [inline openfisca_aotearoa/tests/social_security/sole_parent_support/sole_parent_support__split_care.yaml:1]: Default valure?

File: openfisca_aotearoa/tests/social_security/sole_parent_support/sole_parent_support__split_care.yaml [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+          day:2019-01-28:7: 50
+      Tamaiti:
+        age:
+          day:2019-01-28:7: 5
+      Ruru:
+        age:
+          day:2019-01-28:7: 3
+    families:
+      Whanau:
+        principal: Koro
```
Comment (bonjourmauko) [inline openfisca_aotearoa/tests/social_security/sole_parent_support/sole_parent_support__split_care.yaml:78]: I don't get this. The benefit is calculated for P or for the family?

File: openfisca_aotearoa/variables/acts/citizenship/citizenship.py [hunk 5]
```diff
@@ -1,8 +1,9 @@
 """TODO: Add missing doctring."""
-from openfisca_core.periods import DAY, ETERNITY, DateUnit
+from openfisca_core.periods import DateUnit, DAY, ETERNITY
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/citizenship/citizenship.py]:
```suggestion
from openfisca_core.periods import DateUnit
```

File: openfisca_aotearoa/variables/acts/citizenship/citizenship.py [hunk 6]
```diff
@@ -1,8 +1,9 @@
...
 from openfisca_core.variables import Variable
-from openfisca_aotearoa.variables.demographics.residence import days_since_n_years_ago
+
 from openfisca_aotearoa.entities import Person
+from openfisca_aotearoa.variables.demographics.residence import days_since_n_years_ago
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/citizenship/citizenship.py:6]:
```suggestion
from openfisca_aotearoa.variables.demographics import residence
```

File: openfisca_aotearoa/variables/acts/oranga_tamariki/interpretation/child.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 """TODO: Add missing doctring."""
 
-from openfisca_core import periods, variables
 from openfisca_core import holders
+from openfisca_core import periods, variables
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/oranga_tamariki/interpretation/child.py:4]:
```suggestion
from openfisca_core import holders, periods, variables
```

File: openfisca_aotearoa/variables/acts/oranga_tamariki/interpretation/child.py [hunk 8]
```diff
@@ -1,7 +1,7 @@
 """TODO: Add missing doctring."""
 
-from openfisca_core import periods, variables
 from openfisca_core import holders
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/oranga_tamariki/interpretation/child.py:3]:
```suggestion
```

File: openfisca_aotearoa/tests/social_security/sole_parent_support/sole_parent_support__split_care.yaml [hunk 9]
```diff
@@ -0,0 +1,303 @@
...
```
Comment (bonjourmauko) [inline openfisca_aotearoa/tests/social_security/sole_parent_support/sole_parent_support__split_care.yaml:28]: Here as you're using _social_security__principal_caregiver_ you could instead declare Koro as parent, no?

File: openfisca_aotearoa/variables/acts/social_security/sole_parent_support/sole_parent_support.py [hunk 10]
```diff
@@ -105,33 +118,207 @@ class sole_parent_support__age_threshold(variables.Variable):
...
-    label = "Income is below Sole Parent Support threshold?"
-    definition_period = periods.MONTH
+    label = "At least one dependent child meets age requirement"
+    definition_period = periods.WEEKDAY
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783167"
+
+    def formula_2018_11_26(persons, period, parameters):
+        child_age_threshold = parameters(period).social_security.sole_parent_support.child_age_threshold
+
+        dc = persons.family.members("social_security__dependent_child", period.first_week) * \
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/sole_parent_support/sole_parent_support.py]: What's the corresponding clausde ?

File: openfisca_aotearoa/variables/acts/social_security/sole_parent_support/sole_parent_support.py [hunk 11]
```diff
@@ -105,33 +118,207 @@ class sole_parent_support__age_threshold(variables.Variable):
...
+            persons.family.members("age", period.first_day) < child_age_threshold
+
+        return persons.family.any(dc, role=entities.Family.CHILD) * (persons.has_role(entities.Family.PRINCIPAL) + persons.has_role(entities.Family.PARENT))
```
Comment (bonjourmauko) [inline openfisca_aotearoa/variables/acts/social_security/sole_parent_support/sole_parent_support.py]: Interesting.

I would have thought you were going to use _social_security__principal_caregiver_ for this:

```suggestion
        principal = people("social_security__principal_caregiver", period)
        children = persons.family.any(dc, role=entities.Family.CHILD)
        parent = persons.has_role(entities.Family.PARENT)
        return principal * children * parent
```

**Review by bonjourmauko — DISMISSED** [review]
Great work! Couple of quetions though.

**Review by MarkCalvert — APPROVED** [review]
Will add requested changes in a future PR

**Pull request merged**